src/jaegertracing/utils/UDPClient.h

```diff
@@ -60,7 +60,7 @@ class UDPClient : public agent::thrift::AgentIf {
             throw std::logic_error(oss.str());
         }
         const auto numWritten = ::write(_socket.handle(), data, size);
-        if (numWritten != size) {
+        if ((unsigned)numWritten != size) {
```

Comment:
Hate to nitpick, but I much prefer `static_cast<unsigned>(numWritten)` because it clarifies the purpose of the cast and is grep-able.